torch_geometric/nn/models/explainer.py

```diff
@@ -114,31 +114,268 @@ def forward(self, mask, *args):
+    :obj:`HeteroData` objects. See :obj:`torch_geometric.nn.to_captum_model`
+    for example usage.
+
+    Args:
+        captum_attrs (tuple[tensor]): The output of attribution methods.
+        mask_type (str): Denotes the type of mask to be created with
+            a Captum explainer. Valid inputs are :obj:`"edge"`, :obj:`"node"`,
+            and :obj:`"node_and_edge"`:
+
+            1. :obj:`"edge"`: :obj:`captum_attrs` contains only edge
+               attributions. The returned tuple has no node attributions and a
+               edge attribution dictionary with key `EdgeType` and value
+               edge mask tensor of shape :obj:`[num_edges]`.
+
+            2. :obj:`"node"`: :obj:`captum_attrs` contains only node
+               attributions. The returned tuple has node attribution dictonary
+               with key `NodeType` and value node mask tensor of shape
+               :obj:`[num_nodes, num_features]` and no edge attribution.
+
+            3. :obj:`"node_and_edge"`: :obj:`captum_attrs` contains only node
+                attributions. The returned tuple contains node attribution
+                dictionary followed by edge attribution dictionary.
+
+        metadata (Metadata): The metadata of the heterogeneous graph.
+    """
+    _raise_on_invalid_mask_type(mask_type)
+    node_types = metadata[0]
+    edge_types = metadata[1]
+    x_attr_dict, edge_attr_dict = None, None
+    captum_attrs = [captum_attr.squeeze(0) for captum_attr in captum_attrs]
+    if mask_type == "node":
+        assert len(node_types) == len(captum_attrs)
+        x_attr_dict = dict(zip(node_types, captum_attrs))
+    elif mask_type == "edge":
+        assert len(edge_types) == len(captum_attrs)
+        edge_attr_dict = dict(zip(edge_types, captum_attrs))
+    elif mask_type == "node_and_edge":
+        assert len(edge_types) + len(node_types) == len(captum_attrs)
+        x_attr_dict = dict(zip(node_types, captum_attrs[:len(node_types)]))
+        edge_attr_dict = dict(zip(edge_types, captum_attrs[len(node_types):]))
+    return x_attr_dict, edge_attr_dict
+
+
+def to_captum(
+    model: torch.nn.Module, mask_type: str = "edge",
+    output_idx: Optional[int] = None, metadata: Optional[Metadata] = None
+) -> Union[CaptumModel, CaptumHeteroModel]:
+    r"""
+    Alias for `to_captum_model`.
+
+    .. warning::
+
+        :obj:`~torch_geometric.nn.to_captum` is deprecated and will
+        be removed in a future release.
+        Use :obj:`torch_geometric.nn.to_captum_model` instead.
+
+    """
+    return to_captum_model(model, mask_type, output_idx, metadata)
+
+
+def to_captum_model(
+    model: torch.nn.Module, mask_type: str = "edge",
+    output_idx: Optional[int] = None, metadata: Optional[Metadata] = None
+) -> Union[CaptumModel, CaptumHeteroModel]:
     r"""Converts a model to a model that can be used for
     `Captum.ai <https://captum.ai/>`_ attribution methods.
 
     .. code-block:: python
```

Comment:
Can we make two code-blocks here? One for homogeneous and one for heterogenous?

Reply:
Done.